File: src/DataBaseManager.java
import java.sql.*;
import java.util.ArrayList;

public class DataBaseManager {
    private static final String URL = "jdbc:mysql://localhost:3306/hptldb";
    private static final String USER = "root";
    private static final String PASSWORD = "welcome1234";

    public static Connection getConnection() throws SQLException {
        try {
            Class.forName("com.mysql.cj.jdbc.Driver"); // Load MySQL Driver
        } catch (ClassNotFoundException e) {
            System.out.println("MySQL JDBC Driver not found. Ensure it's added to the classpath.");
            e.printStackTrace();
        }
        return DriverManager.getConnection(URL, USER, PASSWORD);
    }

    // Retrieve all patients using the stored procedure
    public static void getAllPatients() {
        String query = "{CALL GetAllPatients()}";  // Call the stored procedure

        try (Connection conn = getConnection(); CallableStatement stmt = conn.prepareCall(query)) {
            ResultSet rs = stmt.executeQuery();
            while (rs.next()) {
                System.out.println("Patient ID: " + rs.getInt("Patient_ID") + 
                                   ", Name: " + rs.getString("Patient_Name") + 
                                   ", Building: " + rs.getString("Building") + 
                                   ", Room: " + rs.getInt("Room_Number"));
            }
        } catch (SQLException e) {
            e.printStackTrace();
        }
    }
    




    public static void main(String[] args) {
        getAllPatients(); // Fetching all patients using the stored procedure
    }
}

File: src/DataBaseManager.py
import mysql.connector

# MySQL Database Configuration
DB_CONFIG = {
    "host": "localhost",
    "user": "root",
    "password": "welcome1234",
    "database": "hptldb"
}

def get_connection():
    """Establish and return a connection to the MySQL database."""
    try:
        conn = mysql.connector.connect(**DB_CONFIG)
        return conn
    except mysql.connector.Error as err:
        print(f"Error: {err}")
        return None

def get_all_patients():
    """Retrieve all patients using the stored procedure GetAllPatients()."""
    query = "CALL GetAllPatients()"
    
    try:
        conn = get_connection()
        if conn is None:
            return
        
        cursor = conn.cursor(dictionary=True)
        cursor.execute(query)

        # Fetch and display results
        patients = cursor.fetchall()
        for patient in patients:
            print(f"Patient ID: {patient['Patient_ID']}, "
                  f"Name: {patient['Patient_Name']}, "
                  f"Building: {patient['Building']}, "
                  f"Room: {patient['Room_Number']}")

        # Close resources
        cursor.close()
        conn.close()

    except mysql.connector.Error as err:
        print(f"SQL Error: {err}")

if __name__ == "__main__":
    get_all_patients()  # Fetching all patients using the stored procedure
